plugins/org.eclipse.n4js.generator.headless/src/org/eclipse/n4js/generator/headless/HeadlessHelper.java

```diff
@@ -68,7 +67,8 @@
 	 *             in error Case.
 	 */
 	public void registerProjects(BuildSet buildSet, FileBasedWorkspace workspace) throws N4JSCompileException {
-		Iterable<URI> projectUris = Iterables.transform(buildSet.getAllProjects(), p -> p.getLocation());
+		Iterable<FileURI> projectUris = Iterables.transform(buildSet.getAllProjects(),
+				p -> (FileURI) p.getSafeLocation());
```

Comment:
The cast to `FileURI` looks suspicious at a first glance. Needs to be double checked.